Shopify/ruby-lsp
# Add infrastructure for keyword documentation

### Motivation

This PR adds the basic infrastructure for #2227, so that we can slowly document keywords and even receive help from the community in doing so.

### Implementation

The idea is to show a short sentence on hover/completion with a link to longer form documentation.

The way I went about this is to add a separate directory with all of the static documentation in Markdown files. Those files are never read by the LSP, we just link to them from the hover markdown. Clicking on the link will open the documentation on the user's editor.

After this infrastructure is merged, the work to document more keywords is simply:

1. Add the related Prism class to the list of `ALLOWED_TARGETS`
2. Register for the event on the dispatcher (e.g.: `:on_next_node_enter`)
3. Define the exact same handler as yield
```ruby
sig { params(node: Prism::SomeKeywordNode).void }
def on_some_keyword_node_enter(node)
  show_keyword_documentation(node.keyword)
end
```
4. Write both the short and long documentation, in `static_docs.rb` and `static_docs/some_keyword.md`, respectively

### Automated Tests

Added tests.

### Manual Tests

1. On this branch, hover over yield
2. Verify you see the documentation
3. Click the read more link
4. Verify you land on the `yield.md` file
